### ISSUE TITLE
0.5 rc1: Savegame with crash after notification of epidemic.

### ISSUE THREAD

**Issue by hulagutten2** [CorsixTH/CorsixTH#868]
https://drive.google.com/file/d/0B_gMIT2wvyj2ZXNVRU0tVk5uVEU/view?usp=sharing
That savegame crashes every time after 1-2 seconds.

Here are savegames that are a bit earlier. (same minute and one minute before)
https://drive.google.com/file/d/0B_gMIT2wvyj2eFJFR1B0ZGtPaXM/view?usp=sharing

and

https://drive.google.com/file/d/0B_gMIT2wvyj2UUhXcmE5VElLSzg/view?usp=sharing

**Comment by sadger** [comment]
Do you have the output of your gamelog.txt on each of the crashes (if different in each case)?

**Comment by sadger** [comment]
On testing, doesn't seem to be related to epidemics per se, seems to be an infinite walk loop causing the game to freeze, potentially related to #867

**Comment by bobcanthelpyou** [comment]
In Autosave3, Autosave4 and Autosave5 a patient with a hat and Uncommon Cold try to leave the pharmacy and crashes the game. Strange also: an other patient with Gut Rot was in the pharmacy before, now queuing for an other pharmacy room.

```
An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
./Lua/queue.lua:180: Suspected infinite loop
stack traceback:
	[C]: in function 'error'
	./Lua/app.lua:797: in function <./Lua/app.lua:794>
	./Lua/queue.lua:180: in function 'back'
	./Lua/room.lua:695: in function 'tryToFindNearbyPatients'
	./Lua/room.lua:558: in function 'onHumanoidLeave'
	./Lua/humanoid_actions/walk.lua:352: in function 'on_next_tile_set'
	./Lua/humanoid_actions/walk.lua:150: in function <./Lua/humanoid_actions/walk.lua:137>
	(tail call): ?
	./Lua/entity.lua:218: in function 'tick'
	./Lua/world.lua:1121: in function 'onTick'
	./Lua/app.lua:878: in function <./Lua/app.lua:875>
	(tail call): ?
	./Lua/app.lua:779: in function <./Lua/app.lua:774>
```

**Comment by sadger** [comment]
Stack trace looks similar to #860 - same tryToFindNearbyPatients in trace

**Comment by bobcanthelpyou** [comment]
Just notices: When loading the Autosave4.sav and the patient (with a hat, enter the hospital around 14th of April, which cause a freeze when leaving the pharmacy) is waiting in or in front of the pharmacy. And i put a new hired nurse in an other pharmacy the game freeze with a different stack much more similar to #860 

```
An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
./Lua/room.lua:696: Suspected infinite loop
stack traceback:
	[C]: in function 'error'
	./Lua/app.lua:797: in function <./Lua/app.lua:794>
	./Lua/room.lua:696: in function 'tryToFindNearbyPatients'
	./Lua/room.lua:558: in function 'onHumanoidLeave'
	./Lua/humanoid_actions/walk.lua:352: in function 'on_next_tile_set'
	./Lua/humanoid_actions/walk.lua:150: in function <./Lua/humanoid_actions/walk.lua:137>
	(tail call): ?
	./Lua/entity.lua:218: in function 'tick'
	./Lua/world.lua:1121: in function 'onTick'
	./Lua/app.lua:878: in function <./Lua/app.lua:875>
	(tail call): ?
	./Lua/app.lua:779: in function <./Lua/app.lua:774>
```

**Comment by bobcanthelpyou** [comment]
Just notices: When loading the Autosave4.sav and the patient (with a hat, enter the hospital around 14th of April, which cause a freeze when leaving the pharmacy) is waiting in or in front of the pharmacy. And i put a new hired nurse in an other pharmacy the game freeze with a different stack much more similar to #860 

```
An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
./Lua/room.lua:696: Suspected infinite loop
stack traceback:
	[C]: in function 'error'
	./Lua/app.lua:797: in function <./Lua/app.lua:794>
	./Lua/room.lua:696: in function 'tryToFindNearbyPatients'
	./Lua/room.lua:558: in function 'onHumanoidLeave'
	./Lua/humanoid_actions/walk.lua:352: in function 'on_next_tile_set'
	./Lua/humanoid_actions/walk.lua:150: in function <./Lua/humanoid_actions/walk.lua:137>
	(tail call): ?
	./Lua/entity.lua:218: in function 'tick'
	./Lua/world.lua:1121: in function 'onTick'
	./Lua/app.lua:878: in function <./Lua/app.lua:875>
	(tail call): ?
	./Lua/app.lua:779: in function <./Lua/app.lua:774>
```

**Comment by Alberth289346** [comment]
The code jumps back and forth between code in different files (rooms, patients, queues) within the loop. At some point the infinite loop detection triggers, it's mostly a matter of luck at what point exactly that happens.

**Comment by hulagutten2** [comment]
@Alberth289346 Yes, it seems to be a duplicate. I am sorry. I will keep it open, since it has additional save game, but please close this one if that is more appropriate.

**Comment by TheCycoONE** [comment]
Fixed by #871
